route53/records_test.go

```diff
@@ -176,3 +267,135 @@ func TestCreateRecord(t *testing.T) {
+	r := Route53{
+		Service: newmockRoute53Client(t, nil),
+		Domains: map[string]common.Domain{
+			"hyper.converged": common.Domain{
+				CertArn: "arn:aws:acm::12345678910:certificate/111111111-2222-3333-4444-555555555555",
+			},
+		},
+	}
+
+	// test success
+	expected := []*route53.ResourceRecordSet{&testResourceRecordSet, &testResourceRecordSet1, &testResourceRecordSet2}
+	out, err := r.ListRecords(context.TODO(), testHostedZoneID)
+	if err != nil {
+		t.Errorf("expected nil error, got: %s", err)
+	}
+
+	if !reflect.DeepEqual(out, expected) {
+		t.Errorf("expected %+v, got %+v", expected, out)
+	}
+
+	// * ErrCodeNoSuchHostedZone "NoSuchHostedZone"
+	// No hosted zone exists with the ID that you specified.
+	r.Service.(*mockRoute53Client).err = awserr.New(route53.ErrCodeNoSuchHostedZone, "NoSuchHostedZone", nil)
+	_, err = r.ListRecords(context.TODO(), testHostedZoneID)
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrNotFound {
+			t.Errorf("expected error code %s, got: %s", apierror.ErrNotFound, aerr.Code)
+		}
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+
+	// * ErrCodeInvalidInput "InvalidInput"
+	// The input is not valid.
+	r.Service.(*mockRoute53Client).err = awserr.New(route53.ErrCodeInvalidInput, "InvalidInput", nil)
+	_, err = r.ListRecords(context.TODO(), testHostedZoneID)
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrBadRequest {
+			t.Errorf("expected error code %s, got: %s", apierror.ErrBadRequest, aerr.Code)
+		}
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+
+	// test some other, unexpected AWS error
+	r.Service.(*mockRoute53Client).err = awserr.New("UnknownThingyBrokeYo", "ThingyBroke", nil)
+	_, err = r.ListRecords(context.TODO(), testHostedZoneID)
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrBadRequest {
+			t.Errorf("expected error code %s, got: %s", apierror.ErrBadRequest, aerr.Code)
+		}
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+
+	// test non-aws error
+	r.Service.(*mockRoute53Client).err = errors.New("things blowing up!")
+	_, err = r.ListRecords(context.TODO(), testHostedZoneID)
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrInternalError {
+			t.Errorf("expected error code %s, got: %s", apierror.ErrInternalError, aerr.Code)
+		}
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+}
+
+func TestGetRecord(t *testing.T) {
+	r := Route53{
+		Service: newmockRoute53Client(t, nil),
+		Domains: map[string]common.Domain{
+			"hyper.converged": common.Domain{
+				CertArn: "arn:aws:acm::12345678910:certificate/111111111-2222-3333-4444-555555555555",
+			},
+		},
+	}
+
+	expected := &testResourceRecordSet
+	out, err := r.GetRecord(context.TODO(), testHostedZoneID, "foobar.hyper.converged", "A")
```

Comment:
Is it worth testing when the host name ends in a `.` it should also succeed?

Reply:
probably, ill add it